Products/DataCollector/plugins/DataMaps.py

```diff
@@ -58,11 +60,17 @@ def __repr__(self):
     def __iter__(self):
         return iter(self.maps)
 
-    def append(self, obj):
-        self.maps.append(obj)
+    def append(self, objmap):
+        self._add_map(objmap)
 
     def extend(self, objmaps):
-        self.maps.extend(objmaps)
+        for map in objmaps:
+            self._add_map(map)
+
+    def _add_map(self, objmap):
```

Comment:
Is this necessary in addition to the `plugin_name` that comes in on the rel map?

Reply:
yes, the RelationshipMap and ObjectMap classes should ensure that they contain the correct data.  This covers them so long as datamaps are added through the appropriate API